### ISSUE TITLE
UI: scale VM not able to use same custom offering

### ISSUE THREAD

**Issue by sbrueseke** [apache/cloudstack#7389]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### SUMMARY

You should be able to use the same compute offering for scaling a VM when the offering is a custom offering with variable cpu and memory values. Via api call scaleVirtualMachine it is possible to use the same serviceofferingid with other values via details parameter.

##### STEPS TO REPRODUCE

~~~
Create a custom compute offering. Create a new VM using this offering. Stop VM and click on the scale VM icon.
~~~

##### EXPECTED RESULTS

~~~
You should be able to edit cpu an memory of the VM by using the same custom offering
~~~

##### ACTUAL RESULTS

~~~
Custom offering is not visible in UI.
~~~

**Comment by kiranchavala** [comment]
Hi @sbrueseke 

Could you please let me know what type of compute offering you created, was it fixed, custom constrained, custom unconstrained

![scale2](https://user-images.githubusercontent.com/1401014/229411445-659e04c7-40f9-463e-9c8c-080859d7be0f.png)

I was able to edit cpu and memory of the vm when i used custom constrained offering

![scale](https://user-images.githubusercontent.com/1401014/229411508-1d61e97c-e9f2-4d7b-9883-b6f85e80dd10.png)

**Comment by sbrueseke** [comment]
@kiranchavala I am using a Custom constrained.

![Unbenannt](https://user-images.githubusercontent.com/23243455/229435358-6b57d3b1-0494-46fd-a650-41bf4d153213.PNG)

**Comment by shwstppr** [comment]
@sbrueseke I'm not able to reproduce this. Am I missing anything?

https://user-images.githubusercontent.com/153340/229446067-a114a726-9af5-4de1-a31a-7908cda2742d.mp4

**Comment by sbrueseke** [comment]
@shwstppr thank you very much! Interesting so see. Are you able to use tags for the disk offerings your compute offerings are using? In my test environment I have a primary storage which is using tag local-ssd and a shared storage using the tag shared-linstor. Maybe this is the difference.

**Comment by shwstppr** [comment]
@sbrueseke I'll check that

**Comment by sbrueseke** [comment]
I think I found the root cause of this. It is not an UI related issue, it is an api related one.
When using command listServiceOfferings and parameter virtualmachineid and you provide an id of a VM using a custom offering with diskofferingstrictness=true, the api will not include this offering in the response and so the UI is not able to provide this offering.
My humble opinion is that this is not correct behavior and you should be able to scale cpu and ram even if the used offer in uses strictness for the disk. Since scaleVirtualMachine is able to this using the same offering, listServiceOfferings should also include the used custom offering in the response.

**Comment by sbrueseke** [comment]
@DaanHoogland since this is not an UI related issue, should I edit this issue and correct subject and post or should we close this one and I open a new one with the correct explanation?

**Comment by DaanHoogland** [comment]
> @DaanHoogland since this is not an UI related issue, should I edit this issue and correct subject and post or should we close this one and I open a new one with the correct explanation?

completely your choice @sbrueseke . If you think discussion so far helps whoever solves it and doesn´t obfuscate the "real" issue, edit this one, otherwise open a new one. €0,02

**Comment by shwstppr** [comment]
Fixed in #7398
